stubs/docutils/docutils/utils/__init__.pyi

```diff
@@ -20,16 +19,46 @@ class DependencyList:
 _SystemMessageLevel: TypeAlias = Literal[0, 1, 2, 3, 4]
 
 class Reporter:
+    levels: list[str]
+
     DEBUG_LEVEL: Literal[0]
     INFO_LEVEL: Literal[1]
     WARNING_LEVEL: Literal[2]
     ERROR_LEVEL: Literal[3]
     SEVERE_LEVEL: Literal[4]
 
+    def __init__(
+        self,
+        source: str,
+        report_level: int,
+        halt_level: int,
+        stream: IO[str] | str | bool | None = None,
+        debug: bool = False,
+        encoding: str | None = None,
+        error_handler: str = "backslashreplace",
+    ) -> None: ...
+
     source: str
+    error_handler: str
+    debug_flag: bool
     report_level: _SystemMessageLevel
-    halt_level: _SystemMessageLevel
-    def __getattr__(self, __name: str) -> Incomplete: ...
+    halt_level: int
+    stream: ErrorOutput
+    encoding: str
+    observers: list[Callable[[nodes.system_message], None]]
+    max_level: int
+    def set_conditions(
+        self, category: Any, report_level: int, halt_level: int, stream: IO[str] | None = None, debug: bool = False
+    ) -> None: ...
+    def attach_observer(self, observer: Callable[[nodes.system_message], None]) -> None: ...
+    def detach_observer(self, observer: Callable[[nodes.system_message], None]) -> None: ...
+    def notify_observers(self, message: nodes.system_message) -> None: ...
+    def system_message(self, level: int, message: str, *children: nodes.Node, **kwargs: Any) -> nodes.system_message: ...
```

Comment:
```suggestion
    def system_message(self, level: int, message: str | Exception, *children: nodes.Node, base_node = ..., **kwargs: Unused) -> nodes.system_message: ...
```

* `message` can also be an `Exception`.
* `base_node` is one potential kwarg.
* The other kwargs are ignored.

Reply:
> base_node is one potential kwarg.
> The other kwargs are ignored.

where are you getting this from? looks to me like a couple of different kwarg keys are used, no?

Reply:
I was looking at https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/docutils/utils/__init__.py#l156, but I see that I missed the `attributes = kwargs.copy()` line. Still, it's better to either add the kwargs that are actually used or use `_typeshed.Incomplete` for now as a marker to add those later.